module-telemetry/src/main/java/com/mapbox/maps/module/telemetry/MapTelemetryImpl.kt

```diff
@@ -17,50 +23,84 @@ class MapTelemetryImpl : MapTelemetry {
+  private val telemetryService: TelemetryService
 
   /**
-   * Creates a map telemetry instance using appplication context and access token
+   * Creates a map telemetry instance using application context and access token
    *
    * @param appContext the application context
    * @param accessToken the mapbox access token
    */
   constructor(appContext: Context, accessToken: String) {
     this.appContext = appContext
     this.accessToken = accessToken
-    this.telemetry = MapboxTelemetry(appContext, accessToken, BuildConfig.MAPBOX_EVENTS_USER_AGENT)
-    val telemetryState: TelemetryEnabler.State =
-      TelemetryEnabler.retrieveTelemetryStateFromPreferences()
-    if (TelemetryEnabler.State.ENABLED == telemetryState) {
-      telemetry.enable()
+
+    val eventsServiceOptions = EventsServerOptions(accessToken, BuildConfig.MAPBOX_EVENTS_USER_AGENT, null)
+    this.eventsService = EventsService.getOrCreate(eventsServiceOptions)
+
+    this.telemetryService = TelemetryService.getOrCreate(eventsServiceOptions)
+
+    if (TelemetryUtils.getEventsCollectionState()) {
+      enableTelemetryCollection(true)
     }
   }
 
   /**
-   * Creates a map telemetry instance using telemetry, appplication context and access token
+   * Creates a map telemetry instance using telemetry, application context and access token
    *
-   * @param telemetry the mapbox telemetry
    * @param appContext the application context
    * @param accessToken the mapbox access token
+   * @param eventsService the mapbox EventsServiceInterface
+   * @param telemetryService the mapbox TelemetryService
    */
-  @VisibleForTesting
-  constructor(telemetry: MapboxTelemetry, appContext: Context, accessToken: String) {
+  @VisibleForTesting(otherwise = PRIVATE)
+  internal constructor(appContext: Context, accessToken: String, eventsService: EventsServiceInterface, telemetryService: TelemetryService) {
     this.appContext = appContext
     this.accessToken = accessToken
-    this.telemetry = telemetry
+    this.eventsService = eventsService
+    this.telemetryService = telemetryService
   }
 
   /**
    * Register the app user turnstile event
    */
   override fun onAppUserTurnstileEvent() {
-    val turnstileEvent = AppUserTurnstile(
-      BuildConfig.MAPBOX_SDK_IDENTIFIER,
-      BuildConfig.MAPBOX_SDK_VERSION
-    )
-    turnstileEvent.setSkuId("00") // id MAUS MAPS
-    telemetry.push(turnstileEvent)
-    telemetry.push(MapEventFactory.buildMapLoadEvent(PhoneState(appContext)))
+    eventsService.sendTurnstileEvent(
+      TurnstileEvent(
+        UserSKUIdentifier.MAPS_MAUS,
+        BuildConfig.MAPBOX_SDK_IDENTIFIER,
+        BuildConfig.MAPBOX_SDK_VERSION
+      )
+    ) { eventsServiceError ->
+      eventsServiceError?.let {
+        logE(TAG, "EventsServiceError: $it")
+      }
+    }
+
+    val mapLoadEvent = MapEventFactory.buildMapLoadEvent(PhoneState(appContext))
```

Comment:
a bit curious that in `onAppUserTurnstileEvent` we send turnstile event + additional one; if it's OK - shouldn't we actually send  `mapLoadEvent` if `eventsServiceError != null` above? otherwise the data may look a bit inconsistent..

Reply:
we sent two events (turnstile + map.load) with mme, so please let's not modify it while `mme`->`core telemetry` switch

Reply:
this is the original logic, we shouldn't modify these two events during the migration.